src/decorators.ts

```diff
@@ -48,31 +48,52 @@ export interface PropertyOptions {
 
 function createProperty(
     proto: any, name: string, options?: PropertyOptions): void {
-  const notify = options && options.notify || false;
-  const reflectToAttribute = options && options.reflectToAttribute || false;
-  const readOnly = options && options.readOnly || false;
-  const computed = options && options.computed || '';
-  const observer = options && options.observer || '';
-
-  let type;
-  if (options && options.hasOwnProperty('type')) {
-    type = options.type;
-  } else if (
-      (window as any).Reflect && Reflect.hasMetadata && Reflect.getMetadata &&
-      Reflect.hasMetadata('design:type', proto, name)) {
-    type = Reflect.getMetadata('design:type', proto, name);
-  } else {
-    console.error(
-        `A type could not be found for ${name}. ` +
-        'Set a type or configure Metadata Reflection API support.');
-  }
-
   if (!proto.constructor.hasOwnProperty('properties')) {
     Object.defineProperty(proto.constructor, 'properties', {value: {}});
   }
 
+  const defaultOpts: PropertyOptions = {
+    notify: false,
+    reflectToAttribute: false,
+    readOnly: false
+  };
   const finalOpts: PropertyOptions =
-      {type, notify, reflectToAttribute, readOnly, computed, observer};
+      proto.constructor.properties[name] || defaultOpts;
+
+  if (options) {
+    if (options.notify !== undefined) {
```

Comment:
Actually I think we could just do `Object.assign(finalOpts, options)`? And do we even need the default initialization above? Polymer itself has those defaults.

Reply:
that's true. i did wonder why we had these in the first place. i could just `{...finalOpts, options}` in that case.